Lesson25/Anil/map.js

```diff
@@ -0,0 +1,347 @@
+	},
+	{
+		name: 'Smartwatch',
+		brand: 'Apple',
+		price: 399,
+		model: 'Apple Watch Series 6',
+	},
+	{
+		name: 'Camera',
+		brand: 'Canon',
+		price: 999,
+		model: 'EOS 80D',
+	},
+	{
+		name: 'Gaming Console',
+		brand: 'Microsoft',
+		price: 499,
+		model: 'Xbox Series X',
+	},
+	{
+		name: 'Tablet',
+		brand: 'Samsung',
+		price: 649,
+		model: 'Galaxy Tab S7+',
+	},
+	{
+		name: 'Smart Speaker',
+		brand: 'Amazon',
+		price: 99,
+		model: 'Echo Dot (4th Gen)',
+	},
+	{
+		name: 'Fitness Tracker',
+		brand: 'Fitbit',
+		price: 129,
+		model: 'Charge 4',
+	},
+];
+
+const productNames = array => array.map(obj=> obj = obj.name);
+console.log("productNames: ",productNames(products));
+
+// Exercise 4:
+// Given an array of words, find the length of each word using .map().
+const wordsExercise4 = [
+	'apple',
+	'banana',
+	'car',
+	'dog',
+	'elephant',
+	'flower',
+	'guitar',
+	'house',
+	'island',
+	'jungle',
+];
+
+const wordLengths = array => array.map(x=> x= x.length );
+console.log("wordLengths: ", wordLengths(wordsExercise4));
+
+// Exercise 5:
+// Given an array of student objects with their names and grades, extract an array of only the student names and grade using .map(). E.g. ['Alice A', ...]
+const studentsExercise5 = [
+	{
+		id: 1,
+		name: 'Alice',
+		age: 18,
+		grade: 'A',
+	},
+	{
+		id: 2,
+		name: 'Bob',
+		age: 19,
+		grade: 'B',
+	},
+	{
+		id: 3,
+		name: 'Charlie',
+		age: 20,
+		grade: 'B+',
+	},
+	{
+		id: 4,
+		name: 'David',
+		age: 19,
+		grade: 'A-',
+	},
+	{
+		id: 5,
+		name: 'Emma',
+		age: 18,
+		grade: 'A+',
+	},
+	{
+		id: 6,
+		name: 'Frank',
+		age: 20,
+		grade: 'B',
+	},
+	{
+		id: 7,
+		name: 'Grace',
+		age: 19,
+		grade: 'A',
+	},
+	{
+		id: 8,
+		name: 'Hannah',
+		age: 20,
+		grade: 'A-',
+	},
+	{
+		id: 9,
+		name: 'Isaac',
+		age: 18,
+		grade: 'B+',
+	},
+	{
+		id: 10,
+		name: 'Julia',
+		age: 19,
+		grade: 'B-',
+	},
+];
+
+const namesAndGrades = array => array.map(obj=> obj = `${obj.name} ${obj.grade}` );
+console.log("namesAndGrades: ",namesAndGrades(studentsExercise5));
+
+// Exercise 6:
+// Given an array of numbers, convert each number to a string representation using .map().
+const numbersExercise6 = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10];
+
+const stringNumbers = array => array.map(String);
+console.log("stringNumbers: ",stringNumbers(numbersExercise6));
+
+// Exercise 7:
+// Given an array of sentences, split each sentence into an array of words using .map().
+const sentencesExercise7 = [
+	'The quick brown fox jumps over the lazy dog.',
+	'I love to code and solve problems.',
+	'The sun sets in the west.',
+	'Life is like a box of chocolates.',
+	'The cat meowed loudly.',
+	'She walked along the beach, enjoying the salty breeze.',
+	'The rain poured down, soaking everything in sight.',
+	'He smiled and waved goodbye.',
+	'The mountain peaks were covered in snow.',
+	'The room was filled with laughter and joy.',
+];
+
+const splittedWords = array => array.map(x=> x = x.split(" "));
```

Comment:
You don't need to do assignment in the map function
```suggestion
const splittedWords = array => array.map(string => string.split(" "));
```